internal/transport/http2_client.go

```diff
@@ -749,6 +749,24 @@ func (t *http2Client) NewStream(ctx context.Context, callHdr *CallHdr) (*ClientS
 		callHdr = &newCallHdr
 	}
 
+	// Authority specified in the `CallAuthority` CallOption takes the highest
+	// precedence to determine the :authority header. Any value in Host field of
+	// CallHdr is overwritten. But before overriding, we validate the authority
+	// string against the peer certificates and fail the RPC with `UNAVAILABLE`
+	// status code if either of the condition fails.
+	if callHdr.Authority != "" {
+		auth, ok := t.authInfo.(credentials.AuthorityValidator)
+		if !ok {
+			return nil, &NewStreamError{Err: status.Error(codes.Unavailable, fmt.Sprintf("credentials type %s does not implement the AuthorityValidator interface", t.authInfo.AuthType())), AllowTransparentRetry: false}
+		}
+		if err := auth.ValidateAuthority(callHdr.Authority); err != nil {
+			return nil, &NewStreamError{Err: status.Error(codes.Unavailable, err.Error()), AllowTransparentRetry: false}
```

Comment:
Here and above, do we need to set `AllowTransparentRetry: false` give that the default for boolean is `false`?

Reply:
Also, might be useful to include the the authority value that failed validation here, since there is no guarantee that the error string returned by the credentials contains that.